packages/client/lib/config.ts

```diff
@@ -184,15 +184,25 @@ export class Config {
   }
 
   /**
-   * Returns the directory for storing the client sync data
+   * Returns the directory for storing the client chain data
    * based on syncmode and selected chain (subdirectory of 'datadir')
    */
-  getSyncDataDirectory(): string {
-    const syncDirName = this.syncmode === 'light' ? 'lightchaindata' : 'chaindata'
-    const chain = this.common.chainName()
-    const networkDirName = chain === 'mainnet' ? '' : `${chain}/`
+  getChainDataDirectory(): string {
+    const networkDirName = this.common.chainName()
+    const chainDataDirName = this.syncmode === 'light' ? 'lightchain' : 'chain'
 
-    const dataDir = `${this.datadir}/${networkDirName}ethereumjs/${syncDirName}`
+    const dataDir = `${this.datadir}/${networkDirName}/${chainDataDirName}`
```

Comment:
I think I like this approach, since you now ensure that if someone accidentally uses the same data dir for a different chain than it was initially used, they won't start using the "wrong" data.